codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -199,6 +195,21 @@ private static void encode3to4(
         }
     }
 
+    // package-private for testing
+    static int encodedBufferSize(int len, boolean breakLines) {
+        // Cast len to long to prevent overflow
+        long len43 = ((long) len << 2) / 3;
+
+        // Account for padding
+        long ret = (len43 + 3) & ~3;
+
+        if (breakLines) {
+            ret += len43 / MAX_LINE_LENGTH;
+        }
+
+        return ret < Integer.MAX_VALUE ? (int) ret : Integer.MAX_VALUE;
```

Comment:
as this is the "worse" case size that we need we may still be able to do so. At worse we will later on produce an IndexOutOfBoundsException when filling the buffer. So I think it is ok to just return Integer.MAX_VALUE here.